GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -102,74 +102,69 @@ + (NSString *)libraryDataStoragePath {
-      [NSString stringWithFormat:@"%@/%ld/%@/%@", baseEventPath, (long)target, qosTier, mappingID];
+      [NSString stringWithFormat:@"%@/%ld/%@.%@.%@", [GDTCORFlatFileStorage baseEventStoragePath],
+                                 (long)target, eventID, qosTier, mappingID];
 }
 
-+ (NSArray<NSString *> *)searchPathsWithEventSelector:(GDTCORStorageEventSelector *)eventSelector {
-  NSMutableArray<NSString *> *searchPaths = [[NSMutableArray alloc] init];
-  if (eventSelector.selectedQosTiers && eventSelector.selectedQosTiers.count > 0) {
-    for (NSNumber *qosTier in eventSelector.selectedQosTiers) {
-      NSString *searchPath = [self pathForTarget:eventSelector.selectedTarget
-                                         qosTier:qosTier
-                                       mappingID:eventSelector.selectedMappingID];
-      BOOL isDirectory;
-      if ([[NSFileManager defaultManager] fileExistsAtPath:searchPath isDirectory:&isDirectory]) {
-        if (isDirectory) {
-          [searchPaths addObject:searchPath];
-        }
-      }
++ (NSSet<NSString *> *)pathsForTarget:(GDTCORTarget)target
+                             eventIDs:(nullable NSSet<NSNumber *> *)eventIDs
+                             qosTiers:(nullable NSSet<NSNumber *> *)qosTiers
+                           mappingIDs:(nullable NSSet<NSString *> *)mappingIDs {
+  NSMutableSet<NSString *> *paths = [[NSMutableSet alloc] init];
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+  NSString *targetPath = [NSString
+      stringWithFormat:@"%@/%ld", [GDTCORFlatFileStorage baseEventStoragePath], (long)target];
+  NSError *error;
+  NSArray<NSString *> *dirPaths = [fileManager contentsOfDirectoryAtPath:targetPath error:&error];
+  if (error) {
+    GDTCORLogDebug(@"There was an error reading the contents of the target path: %@", error);
+    return paths;
+  }
+  BOOL checkingIDs = eventIDs != nil && eventIDs.count > 0;
+  BOOL checkingQosTiers = qosTiers != nil && qosTiers.count > 0;
+  BOOL checkingMappingIDs = mappingIDs != nil && mappingIDs.count > 0;
+  BOOL checkingAnything = checkingIDs == NO && checkingQosTiers == NO && checkingMappingIDs == NO;
+  for (NSString *path in dirPaths) {
+    if (checkingAnything) {
+      [paths addObject:path];
+      continue;
     }
-  } else {
-    NSString *searchPath = [self pathForTarget:eventSelector.selectedTarget
-                                       qosTier:nil
-                                     mappingID:eventSelector.selectedMappingID];
-    BOOL isDirectory;
-    if ([[NSFileManager defaultManager] fileExistsAtPath:searchPath isDirectory:&isDirectory]) {
-      if (isDirectory) {
-        [searchPaths addObject:searchPath];
-      }
+    NSString *filename = [path lastPathComponent];
+    NSArray<NSString *> *components = [filename componentsSeparatedByString:@"."];
+    if (components.count != 3) {
+      GDTCORLogDebug(@"There was an error reading the filename components: %@", components);
+      return paths;
```

Comment:
I wonder why we are skipping other files if one of the files has an incorrect name?

Reply:
Good point--perhaps we can just delete those files and continue.